espnet2/fileio/read_text.py

```diff
@@ -104,3 +107,86 @@ def read_label(path: Union[Path, str]) -> Dict[str, List[Union[float, int]]]:
             )
         retval[key] = temp_info
     return retval
+
+
+class RandomTextReader(collections.abc.Mapping):
+    """Reader class for random access to text.
+
+    Examples:
```

Comment:
I could not fully understand this part (why this is needed and where this is used).
Can you add more descriptions here and appropriate places?

Reply:
@DongjiGao can you explain for this one?

Reply:
The RandomTextReader is used to read the unpaired text for UASR training. 
Instead of loading the whole text into memory, it read the scp file which stores the byte offset of each text (similar to kaldi ark) and randomly selects unpaired text from it for training using mmap.  
The first line of scp defines how many digits are used to store byte offset. Each of the following lines indicates the start and end byte of each text. So the text can be randomly selected by its line number.